ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh

```diff
@@ -36,8 +38,15 @@ if [ -f "${remove_resources_by_cli}" ]; then
     sh -x "${remove_resources_by_cli}"
 fi
 
-rg_files="${SHARED_DIR}/resourcegroup ${SHARED_DIR}/resourcegroup_cluster ${SHARED_DIR}/RESOURCE_GROUP_NAME"
-for rg_file in ${rg_files}; do
+rg_files=(
+    "${SHARED_DIR}/resourcegroup"
+    "${SHARED_DIR}/resourcegroup_cluster"
+    "${SHARED_DIR}/resourcegroup_vnet"
+    "${SHARED_DIR}/resourcegroup_nsg"
+    "${SHARED_DIR}/resourcegroup_aks"
```

Comment:
Why we need to introduce these different resource group ? They can not be shared to use one.

Per my OCP experience, we even can put everything in one shared resource group, including vnet, nsg, cluster itself.

Reply:
Or you are doing that on purpose for testing?

Reply:
Yes, it is technically possible to place the cloud resources and the hosted cluster into the same resource group. However, in this job, they are spread across different resource groups to test the functioning of hosted clusters with BYO VNET, subnet, and NSG. This approach simulates the "worst" scenario allowed, where resources are placed in different resource groups, ensuring robust testing. 

Additionally, the management cluster has to be placed in its own resource group because it has a different lifecycle. It is managed by the service provider and needs to persist even after the hosted cluster and the BYO resources are torn down. 

(Ref: https://hypershift-docs.netlify.app/how-to/azure/create-azure-cluster_on_aks/)